ngraph/core/reference/include/ngraph/runtime/reference/deformable_convolution.hpp

```diff
@@ -14,31 +14,54 @@ namespace ngraph
+                                                                   const CoordinateDiff& pads_end,
+                                                                   const int64_t& groups,
+                                                                   const int64_t& deformable_groups)
                 {
                     // this implementation supports 2D deformable convolutions
                     NGRAPH_CHECK(in_shape.size() == 4, "Unsupported input rank: ", in_shape);
-
                     NGRAPH_CHECK(o_shape.size() == 4, "Unsupported offset rank: ", o_shape);
-
                     NGRAPH_CHECK(f_shape.size() == 4, "Unsupported kernel rank: ", f_shape);
 
-                    const auto spatial_dims = in_shape.size() - 2;
-                    NGRAPH_CHECK(strides.size() == spatial_dims,
-                                 "Strides not definied for all and only spatial dimensions");
+                    NGRAPH_CHECK(in_shape[1] % groups == 0,
+                                 "Input channels of data batch input must be evenly divisible by "
+                                 "'groups' attribute");
+                    NGRAPH_CHECK(f_shape[0] % groups == 0,
+                                 "Output channels of filters must be evenly divisible by 'groups' "
+                                 "attribute");
+
+                    const Shape scaled_f_shape = [f_shape](int64_t g) {
+                        Shape shape{f_shape};
+                        shape[1] *= g;
+                        return shape;
+                    }(groups);
+
+                    validate_convolution_parameters(in_shape,
+                                                    scaled_f_shape,
+                                                    out_shape,
+                                                    strides,
+                                                    dilations,
+                                                    pads_begin,
+                                                    pads_end);
 
-                    NGRAPH_CHECK(dilations.size() == spatial_dims,
-                                 "Dilations not defined for all and only spatial dimensions");
+                    const Shape f_spatial_shape{std::next(f_shape.begin(), 2), std::end(f_shape)};
+                    const Shape o_spatial_shape{std::next(o_shape.begin(), 2), std::end(o_shape)};
+                    const Shape out_spatial_shape{std::next(out_shape.begin(), 2),
+                                                  std::end(out_shape)};
 
-                    NGRAPH_CHECK((pads_begin.size() == pads_end.size()) &&
-                                     (pads_begin.size() == spatial_dims),
-                                 "Pads not defined for all and only spatial dimensions");
+                    NGRAPH_CHECK(o_shape[1] == deformable_groups * shape_size(f_spatial_shape) * 2,
+                                 "The channels dimension of deformable values input is not "
```

Comment:
We changed 'deformable values' to 'offsets' in specification, hence these error message should use use 'offsets' to not confuse users.